Doc/library/argparse.rst

```diff
@@ -1525,12 +1525,38 @@ there are no options in the parser that look like negative numbers::
    PROG: error: argument -1: expected one argument
 
 If you have positional arguments that must begin with ``-`` and don't look
-like negative numbers, you can insert the pseudo-argument ``'--'`` which tells
-:meth:`~ArgumentParser.parse_args` that everything after that is a positional
-argument::
+like negative numbers, you can use the pseudo-argument ``'--'``.
+See :ref:`specifying-positional-arguments` below for more details.
+
+
+.. _specifying-positional-arguments:
+
+Specifying positional arguments
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When there is ambiguity in deciding whether an argument is positional or for an
+argument, ``--`` can be used to tell :meth:`~ArgumentParser.parse_args` that
+everything after that is a positional argument::
+
+   >>> parser = argparse.ArgumentParser(prog='PROG')
+   >>> parser.add_argument('-n', nargs='+')
+   >>> parser.add_argument('args', nargs='*')
+
+   >>> # ambiguous, so parse_args assumes it's an option
+   >>> parser.parse_args(['-f'])
+   usage: PROG [-h] [-n N [N ...]] [args ...]
+   PROG: error: unrecognized arguments: -f
 
    >>> parser.parse_args(['--', '-f'])
-   Namespace(foo='-f', one=None)
+   Namespace(args=['-f'], n=None)
+
+   >>> # ambiguous, so the -n option greedily accepts arguments
+   >>> parser.parse_args(['-n', '1', '1', '1'])
```

Comment:
This example would be better if all of the arguments weren't the same. Maybe use `['-n', '1', '2', '3']`.